models/ellipsis_finance/bnb/ellipsis_finance_bnb_underlying_tokens.sql

```diff
@@ -0,0 +1,396 @@
+{{config(alias='underlying_tokens')}}
+
+WITH
+
+arth_usd as (
+    SELECT 
+        '0xaf6b98b5dc17f4a9a5199545a1c29ee427266da4' as pool, 
+        '0' as token_id,
+        LOWER('0x8B02998366F7437F6c4138F4b543EA5c000cD608') as token_address
+    
+    UNION 
+
+    SELECT 
+        '0xaf6b98b5dc17f4a9a5199545a1c29ee427266da4' as pool, 
+        '1' as token_id,
+        LOWER('0xe9e7CEA3DedcA5984780Bafc599bD69ADd087D56') as token_address
+        
+    UNION 
+    
+    SELECT 
+        '0xaf6b98b5dc17f4a9a5199545a1c29ee427266da4' as pool, 
+        '2' as token_id,
+        LOWER('0x8AC76a51cc950d9822D68b83fE1Ad97B32Cd580d') as token_address
+
+    UNION 
+    
+    SELECT 
+        '0xaf6b98b5dc17f4a9a5199545a1c29ee427266da4' as pool, 
+        '3' as token_id,
+        LOWER('0x55d398326f99059fF775485246999027B3197955') as token_address
+), 
```

Comment:
When writing hard coded data, you can use VALUES to make it more readable.
```
WITH
pool_underlying as (
SELECT pool, token_id, token_address
FROM ( VALUES
    ('0xaf6b98b5dc17f4a9a5199545a1c29ee427266da4', 0, '0x8b02998366f7437f6c4138f4b543ea5c000cd608')
    ,('0xaf6b98b5dc17f4a9a5199545a1c29ee427266da4', 1, '0xe9e7cea3dedca5984780bafc599bd69add087d56')
    ,('0xaf6b98b5dc17f4a9a5199545a1c29ee427266da4', 2, '0x8ac76a51cc950d9822d68b83fe1ad97b32cd580d')
    ,('0xaf6b98b5dc17f4a9a5199545a1c29ee427266da4', 3, '0x55d398326f99059ff775485246999027b3197955`)
    , ...
  ) as temp_table (pool, token_id, token_address
)

SELECT 
    'bnb' as blockchain, 
    '1' as version, 
    'ellipsis_finance' as project,
    * 
FROM 
pool_underlying 
```

⚠️ But it looks like we should source this data from on chain events so it's not hardcoded.
Can you try if the `MetaPoolDeployed` event is sufficient to source this information?

Reply:
Currently [only 4 events in `MetaPoolDeployed` decoded table](https://dune.com/queries/1756614?d=1).

contract_address | evt_tx_hash | evt_index | evt_block_time | evt_block_number | A | base_pool | coins | deployer | fee | lp_token | pool
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
0xa5d748a3234a81120df7f23c9ea665587dc8d871 | 0x3bad631f9ed8c8c8dd50a0f17e2a6a209cb0546ec978b7da71794d1bc78ecbaf | 145 | 2022-10-11 16:59 | 22088755 | 50 | 0x160caed03795365f3a589f10c379ffa7d75d4e76 | ["0x874966221020d6ac1aed0e2cfad9cbfee0ba713b","0xaf4de8e872131ae328ce21d909c74705d3aaf452"] | 0xcaec009b7ff4f3387ccf247e795e08e08472b0b1 | 4000000 | 0x2ea2fac4b19114b1c6e038701a57b75c0f699831 | 0x81703ab57417118473aeac1b01805f295eec6992
0xa5d748a3234a81120df7f23c9ea665587dc8d871 | 0x671f5e63074d4b569b6702fe1483128b64bfbbb9a5a311e1b2916073351718db | 347 | 2022-10-31 04:55 | 22642628 | 200 | 0x19ec9e3f7b21dd27598e7ad5aae7dc0db00a806d | ["0x8e1403e82dad6b43f78eb1b4b65a5428b983d282","0x5b5bd8913d766d005859ce002533d4838b0ebbb5"] | 0x638de996db08d52913b05b30b89cb8253f3e8bb9 | 50000000 | 0x7cdbf033099a6ad08333dd534ce3786cd348f3db | 0xbd4431ed9377f916e3bd6127db46debe874db3c6
0xa5d748a3234a81120df7f23c9ea665587dc8d871 | 0x13802cd23b3cb108ef78389603012e40a2cea33a18c1173b28330ee68c31f853 | 231 | 2022-11-06 16:10 | 22826123 | 10 | 0x160caed03795365f3a589f10c379ffa7d75d4e76 | ["0xe9e7cea3dedca5984780bafc599bd69add087d56","0xaf4de8e872131ae328ce21d909c74705d3aaf452"] | 0xfa1f780a446e9cdb4c51b031686bcf8b02f8ed7c | 4000000 | 0xf58a8dd6c6be0f00ab3aa55e573ed27b5e7611d6 | 0x6649b6e64c0a2d0134facc73f6ada31d7f8526c9
0xa5d748a3234a81120df7f23c9ea665587dc8d871 | 0xfa45549927a2055d9bdd018140272b7fdce2f1d3eafd3dd9c1d756eb995cb0a5 | 128 | 2022-11-13 20:43 | 23029681 | 200 | 0x160caed03795365f3a589f10c379ffa7d75d4e76 | ["0xb3c11196a4f3b1da7c23d9fb0a3dde9c6340934f","0xaf4de8e872131ae328ce21d909c74705d3aaf452"] | 0xfd6d004e713e809292007c4f70daf12894a2b394 | 4000000 | 0xa786c56d1071761baac0856efbd728c0858dcd16 | 0x23b60e8f84a75a550ae5c5ef64e075dda823c2c1

There might be untracked/undecoded ellipsis factory address for Dune.